github/github.go

```diff
@@ -323,7 +324,7 @@ func parseRate(r *http.Response) Rate {
 // current rate.
 func (c *Client) Rate() Rate {
 	c.rateMu.Lock()
-	rate := c.rate
+	rate := c.rateLimits[c.mostRecent]
```

Comment:
> What if Rate always returns the core rate (like it did before)

Now that I think about it more, are you sure that's what `Rate` currently does? Doesn't it already always return the latest rate limit from either of two categories, whichever was called last? In other words, this PR might be currently preserving previous behavior.

If you're sure that the current `Rate` only contains core category rate limits, can you show me where that happens and what prevents a call to search from updating rate?

It looks to me that search API calls would update it also. See here:

- https://github.com/google/go-github/blob/81ea1e5cb324404cd87c66d2a1050c7c31a2121b/github/github.go#L350-L352
- https://github.com/google/go-github/blob/81ea1e5cb324404cd87c66d2a1050c7c31a2121b/github/search.go#L157